src/ansys/sound/core/spectral_processing/filter.py

```diff
@@ -0,0 +1,311 @@
+
+        Parameters
+        ----------
+        frf : Field
+            Frequency response function (FRF) as a DPF field.
+        """
+        # Set operator inputs.
+        self.__operator_design.connect(0, frf)
+        self.__operator_design.connect(1, self.__sampling_frequency)
+
+        # Run the operator.
+        self.__operator_design.run()
+
+        # Get the output.
+        self.b_coefficients = list(map(float, self.__operator_design.get_output(0, "vec_double")))
+        self.a_coefficients = list(map(float, self.__operator_design.get_output(1, "vec_double")))
+
+    def process(self):
+        """Filter the signal with the current coefficients."""
+        # Check input signal.
+        if self.signal is None:
+            raise PyAnsysSoundException(
+                f"Input signal is not set. Use {__class__.__name__}.signal."
+            )
+
+        if self.a_coefficients == []:
+            raise PyAnsysSoundException(
+                "Filter's denominator coefficients (a_coefficients) cannot be empty. Use "
+                f"{__class__.__name__}.a_coefficients, or the methods "
+                f"{__class__.__name__}.design_FIR_from_FRF() or "
+                f"{__class__.__name__}.design_FIR_from_FRF_file()."
+            )
+
+        if self.b_coefficients == []:
+            raise PyAnsysSoundException(
+                "Filter's numerator coefficients (b_coefficients) cannot be empty. Use "
+                f"{__class__.__name__}.b_coefficients, or the methods "
+                f"{__class__.__name__}.design_FIR_from_FRF() or "
+                f"{__class__.__name__}.design_FIR_from_FRF_file()."
+            )
+
+        # Set operator inputs.
+        self.__operator_filter.connect(0, self.signal)
+        self.__operator_filter.connect(1, self.b_coefficients)
+        self.__operator_filter.connect(2, self.a_coefficients)
+
+        # Run the operator.
+        self.__operator_filter.run()
+
+        # Get the output.
+        self._output = self.__operator_filter.get_output(0, "field")
+
+    def get_output(self) -> Field:
+        """Get the filtered signal as a DPF field.
+
+        Returns
+        -------
+        Field
+            Filtered signal as a DPF field.
+        """
+        if self._output == None:
+            warnings.warn(
+                PyAnsysSoundWarning(
+                    "Output is not processed yet. "
+                    f"Use the {__class__.__name__}.process() method."
+                )
+            )
+        return self._output
+
+    def get_output_as_nparray(self) -> np.ndarray:
+        """Get the filtered sound as a NumPy array.
+
+        Returns
+        -------
+        numpy.ndarray
+            Filtered sound as a NumPy array.
+        """
+        output = self.get_output()
+
+        if output == None:
+            return np.array([])
+
+        return np.array(output.data)
+
+    def plot(self):
+        """Plot the filtered sound in a figure."""
```

Comment:
Plot the filtered **SIGNAL** in a figure

Reply:
done